src/plugins/intel_cpu/src/nodes/executors/kleidiai/kleidiai_mm.cpp

```diff
@@ -64,37 +65,77 @@ MatMulKleidiAIExecutor::MatMulKleidiAIExecutor(const FCAttrs& attrs,
-
-    const size_t nr = ukernel.get_nr();
-    const size_t kr = ukernel.get_kr();
-    const size_t sr = ukernel.get_sr();
-
-    kai_run_rhs_pack_kxn_f32p8x1biasf32_f32_f32_neon(1,
-                                                     N,
-                                                     K,
-                                                     nr,
-                                                     kr,
-                                                     sr,          // Packing arguments
-                                                     rhs_stride,  // RHS stride
-                                                     rhs,         // RHS
-                                                     bias,        // Bias
-                                                     nullptr,     // Scale
-                                                     rhs_packed,  // RHS packed
-                                                     0,
-                                                     nullptr);
+
+    if (inferPrecision == f32) {
+        const size_t rhsPackedSize = kai_get_rhs_packed_size_rhs_pack_kxn_f32p8x1biasf32_f32_f32_neon(N, K);
+        auto rhsPackedDesc = std::make_shared<CpuBlockedMemoryDesc>(f32, Shape({rhsPackedSize}));
+        rhsPackedMem = std::make_shared<Memory>(context->getEngine(), rhsPackedDesc);
+
+        float* bias = biasMem->getDataAs<float>();
+        float* rhs_packed = static_cast<float*>(rhsPackedMem->getData());
+        float* rhs = static_cast<float*>(packedWeights->getData());
+        const size_t rhs_stride = N * sizeof(float);
+
+        const size_t nr = ukernel_f32.get_nr();
+        const size_t kr = ukernel_f32.get_kr();
+        const size_t sr = ukernel_f32.get_sr();
+
+        kai_run_rhs_pack_kxn_f32p8x1biasf32_f32_f32_neon(1,
+                                                        N,
+                                                        K,
+                                                        nr,
+                                                        kr,
+                                                        sr,          // Packing arguments
+                                                        rhs_stride,  // RHS stride
+                                                        rhs,         // RHS
+                                                        bias,        // Bias
+                                                        nullptr,     // Scale
+                                                        rhs_packed,  // RHS packed
+                                                        0,
+                                                        nullptr);
+    }
+    else if (inferPrecision == i8) {
+        const size_t rhs_native_size_qs8cx = K * N * sizeof(int8_t);
+        const size_t rhs_scales_size = N * sizeof(float);
+        float* rhs_scales = new float[rhs_scales_size];
+        int8_t* rhs_native_qs8cx = new int8_t[rhs_native_size_qs8cx];
```

Comment:
Please use the same approach for memory allocation as in fp32 branch, like: 
```
auto rhsPackedDesc = std::make_shared<CpuBlockedMemoryDesc>(f32, Shape({rhsPackedSize}));
rhsPackedMem = std::make_shared<Memory>(context->getEngine(), rhsPackedDesc);
```
For now this dynamically allocated memory is not deallocated, while Memory object works as a conntainer and free the memory in the dctor.